src/main/java/com/radixdlt/crypto/ECPublicKey.java

```diff
@@ -41,52 +41,24 @@
 	private final Supplier<EUID> uid = Suppliers.memoize(this::computeUID);
 	private final int hashCode;
 
-	@JsonCreator
-	private static ECPublicKey fromPublicKey(byte[] key) throws CryptoException {
-		return new ECPublicKey(key);
+	private ECPublicKey(byte[] key) {
+		this.publicKey = Arrays.copyOf(key, key.length);
+		this.hashCode = computeHashCode();
 	}
 
-	public ECPublicKey(byte[] key) throws CryptoException {
-		try {
-			validatePublic(key);
-			this.publicKey = Arrays.copyOf(key, key.length);
-			// As ECPublicKey instances are commonly used to look up associated
-			// data in HashMaps and/or used as keys in HashMaps, there is minimal
-			// performance benefits, and some costs, in computing this lazily.
-			this.hashCode = computeHashCode();
-		} catch (CryptoException ex) {
-			throw ex;
-		} catch (Exception ex) {
-			throw new CryptoException(ex);
-		}
+	private int computeHashCode() {
+		return Arrays.hashCode(publicKey);
 	}
 
-	private void validatePublic(byte[] publicKey) throws CryptoException {
-		if (publicKey == null || publicKey.length == 0) {
-			throw new CryptoException("Public key is empty");
-		}
-
-		int pubkey0 = publicKey[0] & 0xFF;
-		switch (pubkey0) {
-		case 2:
-		case 3:
-			if (publicKey.length != BYTES + 1) {
-				throw new CryptoException("Public key is an invalid compressed size");
-			}
-			break;
-		case 4:
-			if (publicKey.length != (BYTES * 2) + 1) {
-				throw new CryptoException("Public key is an invalid uncompressed size");
-			}
-			break;
-		default:
-			throw new CryptoException("Public key is an invalid format");
-		}
+	@JsonCreator
+	public static ECPublicKey fromBytes(byte[] key) throws CryptoException {
```

Comment:
I thought we were going to get rid of `CryptoException`? I would like that, and replace with specific purpose `PublicKeyCreationException` (in this case)

Reply:
Done